File: tatto_app_mobile/lib/features/tattoo_generator/domain/entities/tattoo.dart
import 'package:equatable/equatable.dart';

class Tattoo extends Equatable {
  final String imageUrl;
  final String prompt;
  final String style;
  final String outputLocation;
  final String aspectRatio;

  const Tattoo({
    required this.imageUrl,
    required this.prompt,
    required this.style,
    required this.outputLocation,
    required this.aspectRatio,
  });

  @override
  List<Object> get props => [imageUrl, prompt, style, outputLocation, aspectRatio];
}

File: backend/app/routers/tattoo.py
from fastapi import APIRouter, HTTPException
from app.schemas.request import TattooRequest
from app.schemas.response import TattooResponse
from app.services.tattoo_service import TattooService

router = APIRouter()
tattoo_service = TattooService()

@router.post("/generate", response_model=TattooResponse)
async def generate_tattoo(request: TattooRequest):
    try:
        image_url = await tattoo_service.generate_tattoo(
            prompt=request.prompt,
            style=request.style,
            output_location=request.output_location
        )
        return TattooResponse(image_url=image_url)
    except Exception as e:
        raise HTTPException(status_code=500, detail=str(e))
